### DISCUSSION TITLE
Week 10 report

### DISCUSSION THREAD

**Discussion by HowDoIUseGit** [SE-TINF22B6/CookPit#56] [Show and tell]
**Advancements:**
- recipe-maker
- recipe upload
- code quality
- SRS/project feedback

***recipe-maker***
The recipe-maker received its first major update! To ensure easy usage and an appealing look for our users we added styling to our recipe-maker. Instead of entering all ingredients as one huge String, you now enter each ingredient separately. This enables the opportunity to correct the list of ingredients before sending the request, if necessary.
We also optimized our api call/ai prompt to receive a better outcome for our users. We, not only formatted our text so one can read it with much more ease, we also added the amount of each ingredient needed for the recipe.
We have a new feature!
Our users are now able to download the created recipe.
Possible additions for the future could be an option to tell the ai for which amount of people the ingredients should last. (At the moment the recipe is considered for 1 person)
![image](https://github.com/SE-TINF22B6/CookPit/assets/73703863/7a111e3a-6e6d-4770-b8e8-7ac1198e408c)

***recipe upload***
You want to participate to our lovely cooking community? You sure can do (soon)! We now have a completed template which will allow our users to contribute their own recipes and food creations in the near future. For now, we reserve the right to fill our database with recipes for our self. However, once the website goes live, everybody with an cookpit account can create and upload recipes.
The next big step will be the implementation of the data entered in the template to actually fill our database.
<img width="337" alt="CookPit_RecipeUpload" src="https://github.com/SE-TINF22B6/CookPit/assets/73703863/6250750b-84b6-4cdc-8069-794f0abd53d1">

***code quality***
To improve our backend code we restructred it into smaller modules. This allows us to easier extend our code base and
use the same code snippets multiple time, if necessary.

***SRS/project feedback***
We have updated our project and SRS to hopefully fulfill the requirements give by Frau Li.

**Comment by BenjaminB109** [comment]
Dear CookPit Team,
I like the recipe maker. The design looks very good, I also find the list of ingredients very practical and the fact that you can download the recipe is a remarkable idea. Keep up the good work.
